test/readme.md

```diff
@@ -1,110 +1,268 @@
+repository.
 
-> Please note, `FAKE_MONGODB_SERVICE_ID` will no longer be needed with the completion of [NODE-3431](https://jira.mongodb.org/browse/NODE-3431).
+## Manually Testing the Driver
 
-```sh
-npm run check:test
-```
+You may want to manually test changes you have made to the driver. The steps below will walk you through how to create a new Node project that uses your local copy of the Node driver.  You can
+modify the steps to work with existing Node projects.
 
-Take note of the `[ topology type: load-balanced ]` printout from mocha to make sure it picked up the environment as expected.
+1. Navigate to a new directory and create a new Node project by running `npm init` in a terminal and    working through the interactive prompts. A new file named `package.json` will be created for you.
+1. In `package.json`, create a new dependency for `mongodb` that points to your local copy of the driver. For example:
+   ```
+   "dependencies": {
+     "mongodb": "/path-to-your-copy-of-the-driver-repo/node-mongodb-native"
+   }
+   ```
+1. Run `npm install` to install the dependency.
+1. Create a new file that uses the driver to test your changes. See the [MongoDB Node.js Quick Start Repo][node-quick-start] for example scripts you can use.
 
-When you are done testing you can shutdown the haproxy load balancer with:
 
-```sh
-$DRIVERS_TOOLS/.evergreen/run-load-balancer.sh stop
-```
+## Testing with Special Environments
 
-### CSFLE
+In order to test some features, you will need to generate and set a specialized group of environment variables.  The subsections below will walk you through how to generate and set the environment variables for these features.
 
-As long as certain environment variables are present and mongodb-client-encryption is installed, FLE will run with a regular mocha execution: `npm run check:test`.
+We recommend using a different terminal for each specialized environment to avoid the environment variables from one specialized environment impacting the test runs for another specialized environment.
 
-Define the following variables in your environment:
+Before you begin any of the subsections below, clone the [drivers-evergreen-tools repo](https://github.com/mongodb-labs/drivers-evergreen-tools.git).
+
+We recommend creating an environment variable named `DRIVERS_TOOLS` that stores the path to your local copy of the driver-evergreen-tools repo: `export DRIVERS_TOOLS="/path/to/your/copy/of/drivers-evergreen-tools"`.
+
+### Serverless
+
+The following steps will walk you through how to create and test a MongoDB Serverless instance.
+
+1. Create the following environment variables using a command like `export PROJECT="node-driver"`.
+
+   - `PROJECT`
+   - `SERVERLESS_DRIVERS_GROUP`
+   - `SERVERLESS_API_PUBLIC_KEY`
+   - `SERVERLESS_API_PRIVATE_KEY`
+   - `SERVERLESS_ATLAS_USER`
+   - `SERVERLESS_ATLAS_PASSWORD`
+   - `LOADBALANCED`
+
+   TODO:  Explain what these variables represent and how to get their values.
```

Comment:
Maybe add: "If you are an Atlas user the values can be found in your project details. If you are testing for driver work, consult the Evergreen project configuration for these values."